server/data/data-cassandra/src/main/java/org/apache/james/user/cassandra/CassandraUsersRepository.java

```diff
@@ -123,7 +123,7 @@ public User getUserByName(String name) {
             .join();
         return Optional.ofNullable(result.one())
             .map(row -> new DefaultUser(row.getString(REALNAME), row.getString(PASSWORD), row.getString(ALGORITHM)))
-            .filter(user -> user.getUserName().equals(name))
+            .filter(user -> user.getUserName().toLowerCase(Locale.US).equals(name))
```

Comment:
it should really be `user.hasName(name)`, having a User object and handling strings outside of it is pointless